workflow/controller/operator.go

```diff
@@ -678,7 +678,11 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 			// Formula: timeToWait = duration * factor^retry_number
 			timeToWait = baseDuration * time.Duration(math.Pow(float64(retryStrategy.Backoff.Factor), float64(len(node.Children))))
 		}
-		waitingDeadline := lastChildNode.FinishedAt.Add(timeToWait)
```

Comment:
@jessesuen I'd love your eyes on this please.

Reply:
`MaxDuration` currently counts from `StartedAt`: https://github.com/argoproj/argo/pull/2976/files

Reply:
We did a stress test recently. Without proper backoff, up to 30% of the workflows failed due to the pod-deleted error.